src/deadline/job_attachments/_incremental_downloads/_manifest_s3_downloads.py

```diff
@@ -602,3 +637,67 @@ def _download_manifest_paths(
     # to report progress 100% at the end
     if progress_tracker:
         progress_tracker.report_progress()
+
+
+class PathCompatibilityError(JobAttachmentsError):
+    """
+    Exception raised when a path is not compatible with the host operating system.
+
+    This exception is raised when a manifest path cannot be used on the current operating system,
+    such as when a Windows path is used on Linux/macOS or vice versa, or when the path contains
+    invalid characters for the current operating system.
+    """
+
+    def __init__(self, path: str, original_path: str, error_message: Optional[str] = None):
+        """
+        Initialize a PathCompatibilityError.
+
+        Args:
+            path: The absolute path that is incompatible.
+            original_path: The original relative path from the manifest.
+            error_message: Optional additional error details with suggestions for resolution.
+        """
+        self.path = path
+        self.original_path = original_path
+        self.error_message = error_message
+        message = f"Path '{path}' is not compatible with the host operating system."
+        if error_message:
+            message += f" {error_message}"
+
+        message += (
+            " This is a likely a result of the selected storage profile missing file system location names"
+            " required to map the outputs to the local storage"
+        )
+        super().__init__(message)
+
+
+def _validate_path_compatibility(path: str) -> tuple[bool, Optional[str]]:
+    """
+    Validates that a path is compatible with the host operating system.
+
+    Args:
+        path: The absolute path to validate.
+
+    Returns:
+        A tuple containing:
+        - A boolean indicating whether the path is compatible.
+        - An optional string with an error message if the path is not compatible.
+    """
+    # Check for OS-specific compatibility
+    if os.name == "nt":  # Windows
+        # Check for Windows-specific path issues
+        if path.startswith("/"):
+            return (
+                False,
+                "Path starts with '/' which is not compatible with Windows. Windows paths should use drive letters (e.g., 'C:\\').",
+            )
+
+    else:  # Linux/macOS
+        # Check for Windows drive letters in Linux/macOS paths
+        if re.match(r"^[a-zA-Z]:\\", path):
```

Comment:
Paths on Windows can also have a UNC form that looks like `\\server\mount\path\to\file.png`. I remember us running into something similar with path validation before, we should double check the best thing to do, which is hopefully using something from the Python standard library.